homeassistant/components/monarchmoney/entity.py

```diff
@@ -0,0 +1,79 @@
+"""Monarch money entity definition."""
+
+from typedmonarchmoney.models import MonarchAccount, MonarchCashflowSummary
+
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import MonarchMoneyDataUpdateCoordinator
+
+
+class MonarchMoneyEntityBase(CoordinatorEntity[MonarchMoneyDataUpdateCoordinator]):
+    """Base entity for Monarch Money with entity name attribute."""
+
+    _attr_has_entity_name = True
+
+
+class MonarchMoneyCashFlowEntity(MonarchMoneyEntityBase):
+    """Custom entity for Cashflow sensors."""
+
+    def __init__(
+        self,
+        coordinator: MonarchMoneyDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the Monarch Money Entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = (
+            f"{coordinator.subscription_id}_cashflow_{description.key}"
+        )
+        self.entity_description = description
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, "monarch_money_cashflow")},
+            name="Cashflow",
+            suggested_area="Banking/Finance",
+        )
+
+    @property
+    def summary_data(self) -> MonarchCashflowSummary:
+        """Return cashflow summary data."""
+        return self.coordinator.cashflow_summary
+
+
+class MonarchMoneyAccountEntity(MonarchMoneyEntityBase):
+    """Define a generic class for Entities."""
+
+    def __init__(
+        self,
+        coordinator: MonarchMoneyDataUpdateCoordinator,
+        description: EntityDescription,
+        account: MonarchAccount,
+    ) -> None:
+        """Initialize the Monarch Money Entity."""
+        super().__init__(coordinator)
+
+        self.entity_description = description
+        self._account_id = account.id
+        self._attr_attribution = (
+            f"Data provided by Monarch Money API via {account.data_provider}"
+        )
+        self._attr_unique_id = f"{coordinator.subscription_id}_{account.name}_{description.translation_key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, account.id)},
+            name=f"{account.institution_name} {account.name}",
+            entry_type=DeviceEntryType.SERVICE,
+            manufacturer=account.data_provider,
+            model=f"{account.institution_name} - {account.type_name} - {account.subtype_name}",
+            configuration_url=account.institution_url,
+            suggested_area="Banking/Finance",
+        )
+
+    @property
+    def account_data(self) -> MonarchAccount:
+        """Return the account data."""
+        if account := self.coordinator.get_account_for_id(self._account_id):
+            return account
+
+        raise ValueError("Account not found")
```

Comment:
Let's implement `available` and if the account id is not in the coordinator we return False and make the entity unavailable